core/src/main/java/org/infinispan/commands/CommandsFactoryImpl.java

```diff
@@ -200,7 +201,8 @@ public void setupDependencies(DataContainer container, CacheNotifier<Object, Obj
                                  OrderedUpdatesManager orderedUpdatesManager, StateTransferLock stateTransferLock) {
       this.dataContainer = container;
       this.notifier = notifier;
-      this.cache = cache;
+      // We shouldn't use encoded entries or anything else here
+      this.cache = AbstractDelegatingCache.unwrapCache(cache);
```

Comment:
FYI, ```advancedCache.withEncoding(IdentityEncoder.class)``` will avoid doing any conversion with the cache. 
Not sure if the test failures are caused by this, but the encoding stuff does not support "naked" caches atm.